torch_xla/core/dynamo_bridge.py

```diff
@@ -391,10 +391,14 @@ def optimized_mod(*args):
 
     # mark_step needs to be blocking since we want to access args's XLADatas
     # and they can't be placeholder.
-    if any(
-        torch_xla._XLAC._check_tensor_need_materialization(
-            [a for a in args if isinstance(a, torch.Tensor)])):
-      xm.mark_step(wait=True)
+    input_tensors_to_sync = [
+        args[i] for i, x in enumerate(
+            torch_xla._XLAC._check_tensor_need_materialization(
+                [a for a in args if isinstance(a, torch.Tensor)])) if x
+    ]
```

Comment:
Feel like these list comprehensions becoming longer and longer haha 😢

Reply:
lol I don't know if this is a python thing, I felt like people tend to write longer python list comprehensions